### PR TITLE
Keybindings: Improve esc / exit / blur logic

### PR DESCRIPTION
**Pull request by torkelo**
When working on DataLink editor we removed the global esc binding. This made esc not work in any input field any longer and found this frustrating as I very frequently use esc to close modals & edit modes and if I happen to have an input focus this does not work. 

This adds back the global esc bind but makes it check the current focused element, if the current focused element is an input, textarea or a slate editor it will blur (defocus it). so next time you hit esc the normal close logic will work. 

It even works with DataLinks / slate editor typeaheads, first esc closes typeahead (handled by editors themselves), second esc will blur, and third will close edit view

### CODE REVIEW

File: public/app/core/services/keybindingSrv.ts [hunk 1]
```diff
@@ -47,9 +47,36 @@ export class KeybindingSrv {
...
+      this.bindGlobal('esc', this.globalEsc);
     }
   }
 
+  globalEsc() {
+    const anyDoc = document as any;
+    const activeElement = anyDoc.activeElement;
+
+    // typehead needs to handle it
+    const typeaheads = $('.slate-typeahead--open');
```
Comment (dprokop) [inline public/app/core/services/keybindingSrv.ts]: I have mixed feelings/don't have strong opinion about this PR. I know the esc to blur the input field was always there, but I don't think it's expected(browser default) behaviour. Especially that we now introduce this logic here which is quite error prone if only somebody changes typeahead's class name.

Also I think the behaviour of the esc should belong to the component interested in it rather that global scope.

Comment (dprokop) [inline public/app/core/services/keybindingSrv.ts]:
```suggestion
    const typeaheads = document. querySelectorAll('.slate-typeahead--open');
```

Comment (torkelo) [reply public/app/core/services/keybindingSrv.ts]: this logic was there before I did not add it. 

I assume ESC works in all model / edit workflows and find it super annoying that it does not anymore. It works in most software & modals so I find removing this was a bad idea.

Comment (dprokop) [reply public/app/core/services/keybindingSrv.ts]: Ah, wait, so when we removed the global bind, we also removed the ESC handling for modals etc? I think we only removed that from the Edit mode

Comment (torkelo) [reply public/app/core/services/keybindingSrv.ts]: Well it disabled it for all modals that have an input field which happens to have focus.

**Review by dprokop — APPROVED** [review]

**Pull request merged**